docs/reference/chisel-releases/slice-definitions.md

```diff
@@ -0,0 +1,446 @@
+
+```yaml
+package: hello
+essential:
+  - hello_copyright
+slices:
+  bins:
+    contents:
+      ...
+  copyright:
+    ...
+```
+
+
+(slice_definitions_format_slices)=
+### `slices`
+
+| Field    | Type              | Required |
+| -------- | ----------------- | -------- |
+| `slices` | `dict(str, dict)` | Required |
+
+Defines the slices of a package. 
+
+The slice names must consist only of lower case letters(`a-z`), digits(`0-9`)
+and minus (`-`) signs. They must be at least three characters long and must start
+with an alphanumeric character.
+
+For example, a slice definition called `data`, for the `ca-certificates` package,
+could look like the following:
+
+```yaml
+slices:
+  data:
+    essential:
+      - openssl_data
+    contents:
+      /etc/ssl/certs/ca-certificates.crt: {text: FIXME, mutable: true}
+      /usr/share/ca-certificates/mozilla/: {until: mutate}
+      /usr/share/ca-certificates/mozilla/**: {until: mutate}
+    mutate: |
+      certs_dir = "/usr/share/ca-certificates/mozilla/"
+      certs = [
+        content.read(certs_dir + path) for path in content.list(certs_dir)
+      ]
+      content.write("/etc/ssl/certs/ca-certificates.crt", "".join(certs))
+```
+
+
+(slice_definitions_format_slices_essential)=
+### `slices.<name>.essential`
+
+| Field       | Type        | Required | Supported values   |
+| ----------- | ----------- | -------- | ------------------ |
+| `essential` | `list(str)` | Optional | An existing slice. |
+
+Lists the slices that are needed and that must be installed before the current slice.
+Slices in this list must be written in their full name
+e.g. `hello_copyright`. This field is similar to
+{ref}`slice_definitions_format_essential`, but only applicable for the current
+slice.
+
+In the following example, `libc6_libs` is a requirement for the `bins`
+slice and must be installed when installing the `bins` slice.
+
+```yaml
+slices:
+  bins:
+    essential:
+      - libc6_libs
+```
+
+
+(slice_definitions_format_slices_contents)=
+### `slices.<name>.contents`
+
+| Field      | Type              | Required |
+| ---------- | ----------------- | -------- |
+| `contents` | `dict(str, dict)` | Optional |
+
+Describes the paths that come from this slice.
+
+```{note}
+Paths must be absolute and must start with `/`.
+
+
+Also, paths can have wildcard characters (`?`, `*` and `**`), where
+ * `?` matches any one character, except for `/`,
+ * `*` matches zero or more characters, except for `/`, and
+ * `**` matches zero or more characters, including `/`.
+```
+
+(slice_definitions_format_slices_contents_copy)=
+### `slices.<name>.contents.<path>.copy`
+
+| Field  | Type  | Required |
+| ------ | ----- | -------- |
+| `copy` | `str` | Optional |
+
+The `copy` field refers to the path Chisel should copy the target path from.
+
+In the following example, Chisel copies the `/bin/original` file from the
+package onto `/bin/moved`.
+
+```yaml
+    contents:
+      /bin/moved:
+        copy: /bin/original
```

Comment:
Shouldn't the format match the convention we use in chisel-releases? e.g. `/bin/moved: {copy: /bin/original}`.

Reply:
That's fair. See 6576c16